src/init/AppRunner.ts

```diff
@@ -56,97 +50,77 @@ export class AppRunner {
-          const val = args[key];
-          if (key !== '_' && Array.isArray(val)) {
-            throw new Error(`Multiple values were provided for: "${key}": "${val.join('", "')}"`);
-          }
-        }
-        return true;
-      })
-      .options({
-        baseUrl: { type: 'string', alias: 'b', requiresArg: true },
-        config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
-        loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
-        mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
-        port: { type: 'number', alias: 'p', default: 3000, requiresArg: true },
-        rootFilePath: { type: 'string', alias: 'f', default: './', requiresArg: true },
-        showStackTrace: { type: 'boolean', alias: 't', default: false },
-        sparqlEndpoint: { type: 'string', alias: 's', requiresArg: true },
-        podConfigJson: { type: 'string', default: './pod-config.json', requiresArg: true },
-      })
-      .parseSync();
+    const params = setupYargvWithBaseArgs(yargs(argv.slice(2))
+      .usage('node ./bin/server.js [args]')).parseSync();
 
-    // Gather settings for instantiating the server
-    const loaderProperties: IComponentsManagerBuilderOptions<App> = {
+    // Gather settings for instantiating VarResolver
+    const loaderProperties = {
       mainModulePath: resolveAssetPath(params.mainModulePath),
       dumpErrorState: true,
       logLevel: params.loggingLevel as LogLevel,
     };
-    const configFile = resolveAssetPath(params.config);
 
-    // Create and execute the app
-    this.createApp(loaderProperties, configFile, params)
-      .then(
-        async(app): Promise<void> => app.start(),
-        (error: Error): void => {
-          // Instantiation of components has failed, so there is no logger to use
-          stderr.write(`Error: could not instantiate server from ${configFile}\n`);
-          stderr.write(`${error.stack}\n`);
-          process.exit(1);
-        },
-      ).catch((error): void => {
-        this.logger.error(`Could not start server: ${error}`, { error });
-        process.exit(1);
-      });
+    // Create varResolver
+    const varConfigFile = resolveAssetPath(params.varConfig);
+    this.createComponent(
+      loaderProperties as IComponentsManagerBuilderOptions<VarResolver>, varConfigFile, {}, varConfigComponentIri,
+    ).then(
+    //  Using varResolver resolve vars and start app
+      async(varResolver): Promise<void> => {
+        let vars: VarRecord;
+        try {
+          vars = await varResolver.handle(argv);
+        } catch (error: unknown) {
+          this.exitWithError(error as Error, 'Error in computing variables', stderr);
+        }
+        return this.initApp(
+          loaderProperties, resolveAssetPath(params.config), vars as unknown as Record<string, any>, stderr,
+        );
+      },
+
+      (error): void => this.exitWithError(
```

Comment:
This one should just be thrown; it will be caught in the level above. That way we don't need to abstract `exitWithError` either.

Reply:
Here, we can provide pinpointed error, with reason. Below catch will loose the distinction of whether error is due to failure in variable-initialization or app-initialization. `exitWithError` currently used in few other places too.

Reply:
> exitWithError currently used in few other places too.

Yes, and all of them should just throw.

> Otherwise below catch will loose the distinction of whether error is due to failure in variable-initialization or app-initialization.

We can still do that; just catch the error and rethrow.